chain/src/facets/StockFacet.sol

```diff
@@ -340,4 +343,72 @@ contract StockFacet {
         );
         TxHelper.createTx(TxType.STOCK_TRANSFER, transferData);
     }
+
+    /// @notice Cancel stock from a stakeholder
+    /// @dev Only OPERATOR_ROLE can cancel stock
+    /// @param security_id The ID of the security to cancel
+    /// @param quantity The quantity of shares to cancel
+    function cancelStock(bytes16 security_id, uint256 quantity) external {
+        Storage storage ds = StorageLib.get();
+
+        if (!AccessControl.hasOperatorRole(msg.sender)) {
+            revert AccessControl.AccessControlUnauthorized(msg.sender, AccessControl.OPERATOR_ROLE);
+        }
+
+        if (quantity == 0) {
+            revert InvalidCancellationQuantity(security_id, quantity);
+        }
+
+        StockActivePosition storage position = ds.stockActivePositions.securities[security_id];
+        if (position.quantity == 0) {
+            revert ZeroQuantityPosition(security_id);
+        }
+
+        if (position.quantity < quantity) {
+            revert InsufficientSharesForCancellation(security_id, quantity, position.quantity);
+        }
+
+        // Get stock class for share tracking
+        uint256 stockClassIdx = ds.stockClassIndex[position.stock_class_id] - 1;
+        StockClass storage stockClass = ds.stockClasses[stockClassIdx];
+        bytes16 balance_security_id = bytes16(0);
+
+        // Handle partial cancellation
+        if (quantity < position.quantity) {
```

Comment:
where's the stock issuance object created for the remainder? you can't just create a balance security id without the actual issuance object

Reply:
Good catch! it's been updated.